### PR TITLE
Feature reply to comment

### PR DESCRIPTION
**Pull request by adisen**
A user can reply to a comment

### CODE REVIEW

File: backend/lib/endpoints/posts.js [hunk 1]
```diff
@@ -29,6 +29,9 @@ async function routes(app) {
   const UNLOGGED_POST_SIZE = 120;
   const EXPIRATION_OPTIONS = ["day", "week", "month"];
 
+  // @desc    Get all posts
```
Comment (robinv85) [inline backend/lib/endpoints/posts.js:32]: not sure why you added this now, seems out of scope of the pull request and it might cause more merge conflicts with other PR's 
not saying this is bad but maybe it's better to open a separate issue for this and look at the possibilities to improve code documentation instead of just adding it

Comment (robinv85) [reply backend/lib/endpoints/posts.js:32]: and if we decide to use this we could add everywhere in the API
also maybe we should chedk the integration with the `fastify-oas` plugin because this can help improve maybe also the API documentation that is automatically generated and can be seen with the Swagger UI?

Comment (adisen) [reply backend/lib/endpoints/posts.js:32]: Sorry. I already created a branch for the documentation. Didn't know it'll cause issues here.

**Comment by adisen** [comment]
> would like to get feedback about the code documentation, what do other engineers think @joshmorel @BMariscal @Naraujo13 @joaofnds
> 
> Also, not sure if this is the best way to use `POST /api/posts/:postId/comments/:commentId/replies` endpoint since this was implemented already before with `POST /api/posts/:postId/comments/` and using `parentId` in the request body, this seems better to have not too complex API routes and offers more flexibility like possible to add more nested levels in the future without adding another level in the route...?

Don't get what you mean here @robinv85

**Comment by adisen** [comment]
I sort of just picked up an issue. That means the issue was not necessary initially then.

**Comment by robinv85** [comment]
@adisen I will try to explain again so it's more clear

so you added this endpoint
`POST /api/posts/:postId/comments/:commentId/replies`

and I thought this functionality already existed before, i also worked on it and I know that @Naraujo13 and @joaofnds have worked on the updated data model and the API but i'm not sure about current atatus, what is working and what still needs to be done

so what I think we had before was more simple, with the same result

`POST /api/posts/:postId/comments/`

- create a comment on a post
- create a reply to a comment, which is also a comment, but includes a parent id in the request body

this way, if you want more than 1 level of nesting you can also use this, but you don't need to have the long URL like `comment/id/replies/id/replies/id...` etc.

**Comment by adisen** [comment]
> @adisen I will try to explain again so it's more clear
> 
> so you added this endpoint
> `POST /api/posts/:postId/comments/:commentId/replies`
> 
> and I thought this functionality already existed before, i also worked on it and I know that @Naraujo13 and @joaofnds have worked on the updated data model and the API but i'm not sure about current atatus, what is working and what still needs to be done
> 
> so what I think we had before was more simple, with the same result
> 
> `POST /api/posts/:postId/comments/`
> 
> * create a comment on a post
> * create a reply to a comment, which is also a comment, but includes a parent id in the request body
> 
> this way, if you want more than 1 level of nesting you can also use this, but you don't need to have the long URL like `comment/id/replies/id/replies/id...` etc.
> 
> @adisen is there a github issue for this? we should make sure that we avoid confusion about what still needs to be done and what is already implemented

Yea there was, I guess we can close this PR?

**Comment by Naraujo13** [comment]
> @adisen I will try to explain again so it's more clear
> 
> so you added this endpoint
> `POST /api/posts/:postId/comments/:commentId/replies`
> 
> and I thought this functionality already existed before, i also worked on it and I know that @Naraujo13 and @joaofnds have worked on the updated data model and the API but i'm not sure about current atatus, what is working and what still needs to be done
> 
> so what I think we had before was more simple, with the same result
> 
> `POST /api/posts/:postId/comments/`
> 
> - create a comment on a post
> - create a reply to a comment, which is also a comment, but includes a parent id in the request body
> 
> this way, if you want more than 1 level of nesting you can also use this, but you don't need to have the long URL like `comment/id/replies/id/replies/id...` etc.
> 
> @adisen is there a github issue for this? we should make sure that we avoid confusion about what still needs to be done and what is already implemented
> 

You're right @robinv85 , the endpoint `POST /posts/:postId/comments/` is used to create comments, nested or not. The nesting is done through the attribute `parentId` and though we currently use only 1 level, we could add more in the future with this structure.

This endpoint is already working since PR #378

**Comment by robinv85** [comment]
@adisen @Naraujo13 is #551 the ticket that is already implemented or not? https://github.com/FightPandemics/FightPandemics/issues/551

**Comment by Naraujo13** [comment]
> @adisen @Naraujo13 is #551 the ticket that is already implemented or not? 

Well that ticket didn't exist when I implemented this feature on #378, but I think it is the same feature.
